build-tools/packages/build-cli/src/commands/publish/tarballs.ts

```diff
@@ -0,0 +1,226 @@
+import { type Logger, type PackageJson } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import execa from "execa";
+import { Gunzip } from "fflate";
+import globby from "globby";
+import latestVersion from "latest-version";
+import { BaseCommand, getTarballName } from "../../library/index.js";
+
+interface TarballMetadata {
+	name: string;
+	version: string;
+	filePath: string;
+	fileName: string;
+}
+
+/**
+ * Publishes a tarball to the package registry unless the version is already published.
+ */
+export default class PublishTarballCommand extends BaseCommand<typeof PublishTarballCommand> {
+	static readonly summary =
+		"Publishes tarballs to the package registry unless the version is already published.";
+
+	static readonly description =
+		"Used to publish a portion of tarballs from a folder based on an input file. The file can contain package names or tarball names.";
+
+	static readonly flags = {
+		dir: Flags.directory({
+			description:
+				"A directory containing tarballs to publish. Tarballs must have the file extension tgz.",
+			exists: true,
+			default: ".",
+		}),
+		orderFile: Flags.file({
+			description:
+				"A file with package names that should be published. Such files can be created using `flub list`.",
+			exists: true,
+			required: true,
+		}),
+		tarball: Flags.boolean({
+			description:
+				"Use this flag to indicate that the orderFile contains tarball names instead of package names. Such files can be created using `flub list --tarball`. This option is deprecated and for backwards compatibility only.",
+			dependsOn: ["orderFile"],
+			deprecated: {
+				message: "This option is deprecated and for backwards compatibility only.",
+			},
+		}),
+		retry: Flags.integer({
+			description: `Number of times to retry publishing a package that fails to publish.`,
+			default: 0,
+		}),
+		dryRun: Flags.boolean({
+			aliases: ["dry-run"],
+			description:
+				"Does everything except publish to the registry. This flag will be passed to 'npm publish'.",
+			default: false,
+		}),
+		access: Flags.string({
+			description: "This flag will be passed to 'npm publish'.",
+			options: ["public", "restricted"],
+		}),
+		publishArgs: Flags.string({
+			description:
+				"This string will be passed to 'npm publish' verbatim. Use this to pass additional custom args to npm publish like --tag.",
+		}),
+	};
+
+	public async run(): Promise<void> {
+		const {
+			access,
+			dir,
+			dryRun,
+			orderFile,
+			retry,
+			tarball: orderFileIsTarballs,
+			publishArgs: rawPublishArgs,
+		} = this.flags;
+		const publishArgs: string[] = (rawPublishArgs ?? "").split(" ");
+		if (access !== undefined) {
+			publishArgs.push("--access", access);
+		}
+		if (dryRun) {
+			publishArgs.push("--dry-run");
+		}
+
+		// eslint-disable-next-line unicorn/no-await-expression-member
+		const packageOrder = (await readLines(orderFile)).filter((line) => {
+			return line !== "" && line !== undefined;
+		});
```

Comment:
I know you don't favor the no await rule.
`await` is syntactic sugar. So instead use .then. Here you should be able to do:
```typescript
		const packageOrder = await readLines(orderFile).then((lines) => lines.filter((line) => {
			return line !== undefined && line !== "";
		}));
```
I also switched undefined and "" checks. It is logical to handle undefined first if needed.

Reply:
Good idea. I tend to gorget about .then and .catch since it's not always convenient.

Reply:
Fixed in latest.